### ISSUE TITLE
Desktop is switched in 10.9 on opening TotalTerminal

### ISSUE THREAD

**Issue by kakysha** [binaryage/totalterminal#78]
While running any application in fullscreen mode (e.g. Chrome) and activating TotalTerminal, the desktops is switched back to the first one (or to that in which the totalterminal was open).

**Comment by chronick** [comment]
I'm still on 10.8, and this is an issue for me as well. I'm not sure if it is fixable given that the totalterminal window is just a normal window (non-fullscreen) with its position modified, although maybe @binaryage has some genius up his sleeve.

**Comment by kakysha** [comment]
On 10.8 totalterminal was accurately sliding down even in fullscreen mode (just above the window, padded a little from top to the height equals to the menu bar)

**Comment by proteus42** [comment]
This is an issue I found with updating to v1.4.2 and 1.4.3.  It seems that you could still get the visor to work over full screen windows with v1.4.1 on Mavericks, just without the animation.  Since the update, I've yet to be able to get the same effect using the old hack.

And @headcanon, have you tried editing the Terminal Info.plist by adding the string LSUIElement or "Application is Agent (UIElement)" with boolean "YES"?  TotalTerminal worked over fullscreen apps with that hack on Mountain Lion.

I think this may just be a bug with the newer version of TotalTerminal as they are trying to include Agent support to the app.

**Comment by darwin** [comment]
Please read #3. There is described a way how to turn Terminal.app in LSUIElement dynamically. As a side-effect this also fixes sliding Visor over fullscreen apps.

**Comment by proteus42** [comment]
Tried all those with no luck.

**Comment by chriskrycho** [comment]
This remains an issue in Yosemite, but only when using the full-screen terminal view.

**Comment by chriskrycho** [comment]
The LSUIElement fix does not appear to be working with full-screen apps in 10.10.

**Comment by cchamberlain** [comment]
@chriskrycho is correct, the LSUIElement will still control the backgrounding of the app as an agent service accessible from the top dock, however it doesn't cause the app to overlay other windows in fullscreen anymore.

@PeacefulChaos I hacked around with my info.plist file last night and was able to get total terminal to overlay windows in full screen.  The combination I used was -

```
<key>LSUIElement</key>
<true/>
<key>LSUIPresentationMode</key>
<string>4</string>
```

I had to restart the laptop before the changes took affect (I suppose killing the process would have worked fine).

Per [Launch Services Keys](https://developer.apple.com/library/mac/documentation/General/Reference/InfoPlistKeyReference/Articles/LaunchServicesKeys.html), this causes the app to hide other UI elements that are in front of its z-position:

4 - All suppressed mode. In this mode, all UI elements are hidden, including the menu bar. UI elements may show themselves automatically in response to mouse movements or other user activity. This option is available only in OS X v10.3 and later.

@binaryage - It would be awesome if this could get rolled into the current LSUIElement setting for users on Yosemite.

**Comment by darwin** [comment]
@cchamberlain I just looked at your proposal in more detail and unfortunately I won't integrate it into next TotalTerminal release. I don't have a programatic way how to set LSUIPresentationMode and I don't want to modify Terminal's plist.

I have one simple rule: never modify Apple's files on the disk. This could cause a lot of headaches for me on support side. I would modify Terminal's plist and with next OS update the change would be gone. I could probably implement some monitoring and test periodically if the changes I made are still there, but it seems like a lot of work.

I think for now, people just have to do it manually like you did. This will help them to be aware of shortcomings of the solution.

Let's wait for WWDC and test beta of 10.11 system. Maybe this behaviour will be extended/changed again and let's hope we will get a programatic solution.
